Makefile

```diff
@@ -1,19 +1,30 @@
 GO := go
 BIN_DIR := bin
-BIN := $(BIN_DIR)/pmg
+BIN_NAME := pmg
 SHELL := /bin/bash
 GITCOMMIT := $(shell git rev-parse HEAD)
 VERSION := "$(shell git describe --tags --abbrev=0)-$(shell git rev-parse --short HEAD)"
 
 GO_CFLAGS=-X 'github.com/safedep/pmg/cmd/version.commit=$(GITCOMMIT)' -X 'github.com/safedep/pmg/cmd/version.version=$(VERSION)'
 GO_LDFLAGS=-ldflags "-w $(GO_CFLAGS)"
 
-.PHONY: all
+# Detect current platform
+GOOS := $(shell go env GOOS)
+GOARCH := $(shell go env GOARCH)
+
+# Set extension for Windows
+ifeq ($(GOOS),windows)
+	EXT := .exe
+else
+	EXT :=
+endif
+
+.PHONY: all clean test pmg create_bin
```

Comment:
@Sahilb315 you've tested this on some windows machine right ?
just want to ensure version flags are properly set